File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
	string makeGood(string s) {
		stack<char>st;
		for (auto i : s)
		{
			if (st.empty())
				st.push(i);
			else
			{
				if (st.top() != i && (st.top() == tolower(i) || tolower(st.top()) == i))
					st.pop();
				else
					st.push(i);
			}
		}
		string res;
		while (!st.empty())
		{
			res.push_back(st.top());
			st.pop();
		}
		reverse(res.begin(), res.end());
		return res;
	}
};

File: 1544-make-the-string-great/1544-make-the-string-great.go
func check(a, b int8) bool {
	x := a - b
	if x < 0 {
		x *= -1
	}
	return x == 32
}

func makeGood(s string) string {
	stack := make([]byte, 0, len(s))

	for i := 0; i < len(s); i++ {
		if len(stack) > 0 && check(int8(stack[len(stack)-1]), int8(s[i])) {
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, s[i])
		}
	}
	return string(stack)
}